lib/pages/getStaticProps/redirects.js

```diff
@@ -15,22 +41,105 @@ pages.forEach(({ route, dataRoute, srcRoute }) => {
   const conditions = ["Cookie=__prerender_bypass,__next_preview_data"];
   const target = `/.netlify/functions/${functionName}`;
 
-  // Add one redirect for the page, but only when the NextJS
-  // preview mode cookies are present
-  redirects.push({
-    route,
-    target,
-    force: true,
-    conditions,
-  });
-
-  // Add one redirect for the data route, same conditions
-  redirects.push({
-    route: dataRoute,
-    target,
-    force: true,
-    conditions,
-  });
+  if (nextConfig.i18n) {
+    const { defaultLocale, locales = [] } = nextConfig.i18n;
+    const isDynamic = !!srcRoute;
+    if (!isDynamic) {
+      // Add a redirect for "expected" route to the defaultLocale-prepended route
+      // i.e. /getStaticProps -> /en/getStaticProps
+      if (defaultLocale) {
+        redirects.push({
+          route,
+          target: `/${defaultLocale}${route}`,
+        });
+      }
```

Comment:
We don't need a default for the data route, correct?

Reply:
right. the pages know the correct dataRoute to search for. this takes something like `/about` -> `/en/about` so once you're on `/about` it knows to search for `.../en/about.json`